### PR TITLE
Add Rails 4.2 generated template changes

### PR DESCRIPTION
**Pull request by lizconlan**
Extracted from http://railsdiff.org/4.1.16/4.2.9

Mostly harmless / boilerplate changes except for the `squash!` commits which introduce config or other changes that would benefit from closer attention:

* adding `assets.digest = true` sounds like it might have unintended consequences but I'm not certain
* the `bin/setup` file feels to me like we should replace most of the content with a TODO comment and consider it separately
* `ActiveSupport.to_time_preserves_timezone` looks like it might save us some major headaches down the line
 
The commit message for moving the cookies_serializer config is a bit misleading - pulling it out into its own file was introduced in 4.1 but it appears I opted to configure it inside application.rb instead.

### CODE REVIEW

File: config/initializers/to_time_preserves_timezone.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Be sure to restart your server when you modify this file.
+
+# Preserve the timezone of the receiver when calling to `to_time`.
+# Ruby 2.4 will change the behavior of `to_time` to preserve the timezone
+# when converting to an instance of `Time` instead of the previous behavior
+# of converting to the local system timezone.
+#
+# Rails 5.0 introduced this config option so that apps made with earlier
+# versions of Rails are not affected when upgrading.
+ActiveSupport.to_time_preserves_timezone = true
```
Comment (lizconlan) [inline config/initializers/to_time_preserves_timezone.rb]: Rails told me to do it!

File: config/environments/development.rb [hunk 2]
```diff
@@ -48,6 +48,8 @@
     config.action_mailer.delivery_method = :sendmail
   end
 
+  config.web_console.whitelisted_ips = '10.10.10.0/16'
```
Comment (lizconlan) [inline config/environments/development.rb]: this isn't from Railsdiff but it felt like a related place to put it

File: config/initializers/cookies_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Set the cookie serializer to :hybrid to migrate the old format Marshalled
+# cookies to the new, more secure, JSON format
+Rails.application.config.action_dispatch.cookies_serializer = :hybrid
```
Comment (garethrees) [inline config/initializers/cookies_serializer.rb]: No `\n` at end of file.

File: config/initializers/to_time_preserves_timezone.rb (hunk 1 above)
Comment (garethrees) [inline config/initializers/to_time_preserves_timezone.rb]: > Rails told me to do it!

?

See https://robots.thoughtbot.com/no-newline-at-end-of-file

**Comment by garethrees** [comment]
I had this notification in email but doesn't look like we're setting the web_console value now? Guessing thats intentional?

> > @@ -48,6 +48,8 @@
>    config.action_mailer.delivery_method = :sendmail
>  end
> 
> +  config.web_console.whitelisted_ips = '10.10.10.0/16'
> this isn't from Railsdiff but it felt like a related place to put it

File: config/initializers/to_time_preserves_timezone.rb (hunk 1 above)
Comment (lizconlan) [inline config/initializers/to_time_preserves_timezone.rb]: Bizarrely http://railsdiff.org/4.1.16/4.2.9 (scroll down to 'config/initializers/cookies_serializer.rb'), taking it out though

**Comment by lizconlan** [comment]
> I had this notification in email but doesn't look like we're setting the web_console value now? Guessing thats intentional?

I think I missed cherry-picking it when I made the new branch. Will add it back rather than have yet another pull request just for that

**Comment by lizconlan** [comment]
[ waiting for spec run ]

**Comment by lizconlan** [comment]
no new fails \o/

File: config/initializers/to_time_preserves_timezone.rb (hunk 1 above)
Comment (garethrees) [inline config/initializers/to_time_preserves_timezone.rb]: To me it looks like 4.2.x is [_adding_](https://github.com/rails/rails/commit/bb63e9c5902f4f6374e9c3fa5bd8041d7be9fa91) the final newline.

A final newline is not the same as a final **blank** line.

**Pull request merged**